src/main/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -57,10 +57,12 @@
     public final static String INCORRECT_HEADER_SIZE_MSG = "Incorrect header size for file: ";
     public final static String UNEXPECTED_BLOCK_LENGTH_MSG = "Unexpected compressed block length: ";
     public final static String PREMATURE_END_MSG = "Premature end of file: ";
-    public final static String CANNOT_SEEK_STREAM_MSG = "Cannot seek on stream based file ";
+    public final static String CANNOT_SEEK_CLOSED_STREAM_MSG = "Cannot seek a position for a closed stream";
+    public final static String CANNOT_SEEK_NON_FILE_STREAM_MSG = "Cannot seek a position for a non-file stream";
     public final static String INVALID_FILE_PTR_MSG = "Invalid file pointer: ";
 
     private InputStream mStream = null;
+    private boolean mIsClosed = false;
```

Comment:
Are there other cases where `mIsClosed` could be used? If so, the `CANNOT_SEEK_CLOSED_STREAM_MSG` could be re-named to `CLOSED_STREAM_MSG` and use the String "Closed stream error: ". Then, inside the seek() method, the message could add the "cannot seek a position".